src/transformers/models/maskformer/modeling_maskformer.py

```diff
@@ -0,0 +1,2492 @@
+            class_queries_logits (`torch.Tensor`):
+                A tensor of shape `batch_size, num_queries, num_classes`
+            mask_labels (`torch.Tensor`):
+                A tensor of shape `batch_size, num_classes, height, width`
+            class_labels (`torch.Tensor`):
+                A tensor of shape `batch_size, num_classes`
+            auxilary_predictions (`Dict[str, torch.Tensor]`, *optional*):
+                if `use_auxilary_loss` was set to `true` in [`MaskFormerConfig`], then it contains the logits from the
+                inner layers of the Detr's Decoder.
+
+        Returns:
+            `Dict[str, Tensor]`: A dict of `torch.Tensor` containing two keys:
+            - **loss_cross_entropy** -- The loss computed using cross entropy on the predicted and ground truth labels.
+            - **loss_mask** -- The loss computed using sigmoid focal loss on the predicted and ground truth masks.
+            - **loss_dice** -- The loss computed using dice loss on the predicted on the predicted and ground truth
+              masks.
+            if `use_auxilary_loss` was set to `true` in [`MaskFormerConfig`], the dictionary contains addional losses
+            for each auxilary predictions.
+        """
+
+        # Retrieve the matching between the outputs of the last layer and the labels
+        indices = self.matcher(masks_queries_logits, class_queries_logits, mask_labels, class_labels)
+
+        # Compute the average number of target masks accross all nodes, for normalization purposes
+        num_masks: Number = self.get_num_masks(class_labels, device=class_labels.device)
+
+        # Compute all the requested losses
+        losses: Dict[str, Tensor] = {
+            **self.loss_masks(masks_queries_logits, mask_labels, indices, num_masks),
+            **self.loss_labels(class_queries_logits, class_labels, indices),
+        }
+
+        # In case of auxiliary losses, we repeat this process with the output of each intermediate layer.
+        if auxilary_predictions is not None:
+            for idx, aux_outputs in enumerate(auxilary_predictions):
+                masks_queries_logits = aux_outputs["masks_queries_logits"]
+                class_queries_logits = aux_outputs["class_queries_logits"]
+                loss_dict = self.forward(masks_queries_logits, class_queries_logits, mask_labels, class_labels)
+                loss_dict = {f"{key}_{idx}": value for key, value in loss_dict.items()}
+                losses.update(loss_dict)
+
+        return losses
+
+    def get_num_masks(self, class_labels: torch.Tensor, device: torch.device) -> torch.Tensor:
+        # Compute the average number of target masks accross all nodes, for normalization purposes
+        num_masks = class_labels.shape[0]
+        num_masks_pt = torch.as_tensor([num_masks], dtype=torch.float, device=device)
+        return num_masks_pt
+
+
+class MaskFormerSwinTransformerBackbone(nn.Module):
```

Comment:
Wondering if this class is necessary, or whether we can directly make `MaskFormerSwinModel` output hidden states of shape (B, C, H, W).